src/harness/testcases/WINNF_FT_S_PCR_testcase.py

```diff
@@ -0,0 +1,354 @@
+  'installationParam'. But this parameter is removed for devices where
+  conditionals are pre-loaded. Adding the 'installationParam' into
+  registrationRequests by taking the corresponding values from
+  conditionalRegistrationData.
+
+  Args:
+    registration_request: A list of individual CBSD registration
+      requests (each of which is itself a dictionary).
+    conditional_registration_data: A list of individual CBSD registration
+      data that need to be preloaded into SAS (each of which is a dictionary).
+      data that need to be preloaded into SAS (each of which is a dictionary).
+      the fccId and cbsdSerialNumber fields are required, other fields are
+      optional but required for ppa reference model.
+
+  Raises:
+    It will throws an exception if the installationParam object and required
+      fields is not found in conditionalRegistrationData and registrationRequests
+      for category B.
+
+  """
+  for device in registration_request:
+    if 'installationParam' not in device:
+      install_param_assigned = False
+      for conditional_params in conditional_registration_data:
+        # Check if FCC_ID+Serial_Number present in registrationRequest
+        # and conditional_params match and add the 'installationParam'.
+        if (conditional_params['fccId'] == device['fccId'] and
+                conditional_params['cbsdSerialNumber'] == device['cbsdSerialNumber']):
+          device.update({'installationParam': conditional_params['installationParam']})
+          install_param_assigned = True
+          # If the cbsdCategory is not present in registration request then
+          # assign it to the cbsdCategory in conditional_params.
+          if 'cbsdCategory' not in device:
+            device.update({'cbsdCategory': conditional_params.get('cbsdCategory',
+                                                                  'B')})
+      # Raise an exception if InstallationParam is not found in the conditionals.
+      if not install_param_assigned:
+        raise Exception("InstallationParam not found in conditionals for device "
+                        "%s:%s" % (device['fccId'], device['cbsdSerialNumber']))
+
+
+class PpaCreationTestcase(sas_testcase.SasTestCase):
+  """Implementation of PCR tests.
+
+  Checks the area of the non-overlapping difference between the maximum PPA boundary
+  created by SAS UUT shall be no more than 10% of the area of the maximum PPA boundary
+  created by the Reference Model for different varying condition and verify the SAS UUT
+  is able to create PPA zone or error.
+  """
+
+  @classmethod
+  def setUpClass(cls):
+    """Verify the reference model configuration is proper for the test setup."""
+    # Verify reference model configuration by executing the test_config.py to
+    # determine success or failure based on the verification result to proceed the
+    # test case execution.
+
+    test_config_setup_verification = subprocess.check_output(TEST_CONFIG_COMMAND,
```

Comment:
See my global comment: I was not saying that you should use this script which is not written for that purpose at all (just a tool to help people verify that their python configuration is proper). Please check the earlier comment thread.

Please remove this piece of code entirely.

Reply:
Based   on the suggestion we have removed the setupClass and tearDownClass   accordingly